comp/core/tagger/telemetry/telemetry.go

```diff
@@ -0,0 +1,170 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package telemetry defines the telemetry for the Tagger component.
+package telemetry
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/comp/core/tagger/types"
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
+)
+
+const (
+	subsystem = "tagger"
+	// queryEmptyEntityID refers to a query made with an empty entity id
+	queryEmptyEntityID = "empty_entity_id"
+	// queryEmptyTags refers to a query that returned no tags
+	queryEmptyTags = "empty_tags"
+	// querySuccess refers to a successful query
+	querySuccess = "success"
+)
+
+// CardinalityTelemetry contains the telemetry for a specific cardinality level.
+type CardinalityTelemetry struct {
+	EmptyEntityID telemetry.SimpleCounter
+	EmptyTags     telemetry.SimpleCounter
+	Success       telemetry.SimpleCounter
+}
+
+// Store contains all the telemetry for the Tagger component.
+type Store struct {
+	// / StoredEntities tracks how many entities are stored in the tagger.
+	StoredEntities telemetry.Gauge
+	// UpdatedEntities tracks the number of updates to tagger entities.
+	UpdatedEntities telemetry.Counter
+
+	// PrunedEntities tracks the number of pruned tagger entities.
+	PrunedEntities telemetry.Gauge
+
+	// ClientStreamErrors tracks how many errors were received when streaming
+	// tagger events.
+	ClientStreamErrors telemetry.Counter
+
+	// ServerStreamErrors tracks how many errors happened when streaming
+	// out tagger events.
+	ServerStreamErrors telemetry.Counter
+
+	// Subscribers tracks how many subscribers the tagger has.
+	Subscribers telemetry.Gauge
+	// Events tracks the number of tagger events being sent out.
+	Events telemetry.Counter
+
+	// Sends tracks the number of times the tagger has sent a
+	// notification with a group of events.
+	Sends telemetry.Counter
+
+	// Receives tracks the number of times the tagger has received a
+	// notification with a group of events.
+	Receives telemetry.Counter
+
+	LowCardinalityQueries          CardinalityTelemetry
+	OrchestratorCardinalityQueries CardinalityTelemetry
+	HighCardinalityQueries         CardinalityTelemetry
+	UnknownCardinalityQueries      CardinalityTelemetry
+}
+
+var initializeOnce sync.Once
+var store *Store
+
+// GetStore returns the global store.
+func GetStore() *Store {
+	return store
+}
+
+// NewStore returns a new Store.
+func NewStore(telemetryComp telemetry.Component) *Store {
+	initializeOnce.Do(func() {
```

Comment:
Is there a reason why some telemetries are protected against several creations like in this function but not [this one](https://github.com/DataDog/datadog-agent/pull/27195/files#diff-6d7fee24c483e5f6d376186773401fae760198870a26ad533cfc4bbe950bddc9R164) for example?

Reply:
Nice question. I think I had this at first before adding the new `GetTaggerTelemetryStore` which is use the tagger server. Prior to that there could be a case in which we created the same telemetry metric twixe causing a runtime panic. Now that we have it, we can remove it 😄 

Thanks for pointing it out

Reply:
Done https://github.com/DataDog/datadog-agent/pull/27195/commits/8d6e1ad3f59259b07ec0c72b5d0747855f55d756

Reply:
After learning about the system probe setup, I decided to bring the initialize once back. That way we can warranty that we only initialize the metrics once, no matter the context. 

Here is the commit that modified the code to do it https://github.com/DataDog/datadog-agent/pull/27195/commits/48bb66ca3e076f8ff0864744ddbfae7dd080bd41